examples/CMakeLists.txt

```diff
@@ -6,6 +6,17 @@ ELSE()
 	TARGET_LINK_LIBRARIES(cgit2 git2 pthread)
 ENDIF()
 
+# Use -Wno-declaration-after-statement -Wunused-but-set-variable on global.c
+FOREACH(warn declaration-after-statement unused-but-set-variable)
+	SET(no_warn "-Wno-${warn}")
+	STRING(TOUPPER ${warn} _UPCASE)
+	SET(_var "IS_WNO_${_UPCASE}_SUPPORTED")
+	CHECK_C_COMPILER_FLAG(${no_warn} ${_var})
+	IF(${_var})
+		SET_PROPERTY(SOURCE general.c APPEND_STRING PROPERTY COMPILE_FLAGS " ${no_warn}")
+	ENDIF()
+ENDFOREACH()
+
```

Comment:
What do you mean by "give up"? `general.c` includes a series of loosely coupled code snippets. It makes no sense to move all declarations to the top of `main()`. The other ways to silent these warnings include:
- use IF(CMAKE_COMPILER_IS_GNUCC) instead of checking for compiler flag;
- use `#pragma GCC ignored -Wdeclaration-after-statement` inside `general.c`.